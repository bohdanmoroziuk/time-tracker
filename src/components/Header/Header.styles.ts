import { StyleSheet } from 'react-native';

export default StyleSheet.create({
  header: {
    paddingTop: 35,
    paddingBottom: 15,
    borderBottomWidth: 1,
    borderBottomColor: '#D6D7DA',
  },
  title: {
    fontSize: 18,
    fontWeight: 'bold',
    textAlign: 'center',
  },
});
