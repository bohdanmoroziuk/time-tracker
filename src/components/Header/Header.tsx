import { FunctionComponent } from 'react';
import { View, Text} from 'react-native';

import styles from './Header.styles';

export interface HeaderProps {
  title: string;
}

const Header: FunctionComponent<HeaderProps> = ({ title }) => (
  <View style={styles.header}>
    <Text style={styles.title}>
      {title}
    </Text>
  </View>
);

export default Header;
