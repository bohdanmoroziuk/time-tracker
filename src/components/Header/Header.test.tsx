import { render } from '@testing-library/react-native';

import Header from './Header';

describe('Header', () => {
  it('renders successfully', () => {
    const title = "Timers";

    const { toJSON } = render(<Header title={title} />);

    expect(toJSON()).toMatchSnapshot();
  });

  it('renders correctly', () => {
    const title = "Timers";

    const { queryByText } = render(<Header title={title} />);

    expect(queryByText(title)).toBeDefined();
  });
});
