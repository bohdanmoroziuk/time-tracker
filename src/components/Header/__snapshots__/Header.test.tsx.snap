// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders successfully 1`] = `
<View
  style={
    Object {
      "borderBottomColor": "#D6D7DA",
      "borderBottomWidth": 1,
      "paddingBottom": 15,
      "paddingTop": 35,
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 18,
        "fontWeight": "bold",
        "textAlign": "center",
      }
    }
  >
    Timers
  </Text>
</View>
`;
